scripts/ci/enforce_runtime_determinism.py

```
@@ -0,0 +1,196 @@
+    runtime_root = repo / "backend" / "app"
+    entrypoints = [repo / entry for entry in ENTRYPOINTS]
+    reachable = build_reachability(entrypoints)
+
+    violations = scan_runtime(runtime_root, reachable)
+    header = [
+        "Determinism scan",
+        f"Scanned root: {runtime_root}",
+        f"Reachable modules: {len(reachable)}",
+        f"Violations: {len(violations)}",
+    ]
+    output_lines: List[str] = header + [""] + ["Violations:"] + violations
```

Ruff (0.14.11)

Consider `[*header, "", "Violations:", *violations]` instead of concatenation

Replace with `[*header, "", "Violations:", *violations]`

(RUF005)